File: ROS2 Course/activity_2/src/py_pkg/py_pkg/number_publisher.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from example_interfaces.msg import Int64


class NumberPublisher(Node):
    def __init__(self):
        super().__init__("number_publisher")
        self.publisher_ = self.create_publisher(Int64, "number", 10)
        self.timer = self.create_timer(1.0, self.timer_callback)
        self.number = 2


    def timer_callback(self):
        msg = Int64()
        msg.data = self.number
        self.publisher_.publish(msg)
        self.get_logger().info("Publishing " + str(self.number))



def main(args=None):
    rclpy.init(args=args)
    node = NumberPublisher()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: ROS2 Course/activity_3/src/cpp_pkg/src/number_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int64.hpp"

using namespace std::chrono_literals;

class NumberPublisher : public rclcpp::Node {
    public:
        NumberPublisher() : Node("number_publisher"), number_(2)
        {
            publisher_ = this->create_publisher<example_interfaces::msg::Int64>("number", 10);
            timer_ = this->create_wall_timer(1s, std::bind(&NumberPublisher::timer_callback, this));
        }
    private:
        void timer_callback(){
            auto msg = example_interfaces::msg::Int64();
            msg.data = 2;
            RCLCPP_INFO(this->get_logger(), "Publishing %ld", msg.data);
            publisher_->publish(msg);
        }
    int64_t number_;
    rclcpp::TimerBase::SharedPtr timer_;
    rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr publisher_;
};


int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<NumberPublisher>());
  rclcpp::shutdown();
  return 0;
}